htdocs/accountancy/customer/lines.php

```diff
@@ -107,67 +107,96 @@
+	'f.ref' 				=> array('label' => "Invoice", 				'position' => 1, 'checked' => '1', 'enabled' => '1'),
+	'f.datef' 				=> array('label' => "Date", 				'position' => 1, 'checked' => '1', 'enabled' => '1'), // f.datef, f.ref, fd.rowid
+	'p.ref' 				=> array('label' => "ProductRef", 			'position' => 1, 'checked' => '1', 'enabled' => '1'),
+	'fd.description' 		=> array('label' => "ProductDescription", 	'position' => 1, 'checked' => '1', 'enabled' => '1'),
+	'fd.total_ht' 			=> array('label' => "Amount", 				'position' => 1, 'checked' => '1', 'enabled' => '1'),
+	'fd.tva_tx'				=> array('label' => "VATRate", 				'position' => 1, 'checked' => '1', 'enabled' => '1'),
+	's.nom' 				=> array('label' => "ThirdParty", 			'position' => 1, 'checked' => '1', 'enabled' => '1'),
+	'co.label' 				=> array('label' => "Country", 				'position' => 1, 'checked' => '1', 'enabled' => '1'),
+	's.tva_intra' 			=> array('label' => "VATIntra", 			'position' => 1, 'checked' => '1', 'enabled' => '1'),
+	'aa.account_number' 	=> array('label' => "AccountAccounting", 	'position' => 1, 'checked' => '1', 'enabled' => '1'),
+);
+// @phpstan-ignore-next-line
+$arrayfields = dol_sort_array($arrayfields, 'position');
+
 /*
  * Actions
  */
 
-// Purge search criteria
-if (GETPOST('button_removefilter_x', 'alpha') || GETPOST('button_removefilter.x', 'alpha') || GETPOST('button_removefilter', 'alpha')) { // All tests are required to be compatible with all browsers
-	$search_societe = '';
-	$search_lineid = '';
-	$search_ref = '';
-	$search_invoice = '';
-	$search_label = '';
-	$search_desc = '';
-	$search_amount = '';
-	$search_account = '';
-	$search_vat = '';
-	$search_date_startday = '';
-	$search_date_startmonth = '';
-	$search_date_startyear = '';
-	$search_date_endday = '';
-	$search_date_endmonth = '';
-	$search_date_endyear = '';
-	$search_date_start = '';
-	$search_date_end = '';
-	$search_country = '';
-	$search_tvaintra = '';
+$parameters = array('arrayfields' => &$arrayfields);
+$reshook = $hookmanager->executeHooks('doActions', $parameters, $object, $action); // Note that $action and $object may have been modified by some hooks
+if ($reshook < 0) {
+	setEventMessages($hookmanager->error, $hookmanager->errors, 'errors');
 }
 
-if (is_array($changeaccount) && count($changeaccount) > 0 && $user->hasRight('accounting', 'bind', 'write')) {
-	$error = 0;
-
-	if (!(GETPOSTINT('account_parent') >= 0)) {
-		$error++;
-		setEventMessages($langs->trans("ErrorFieldRequired", $langs->transnoentitiesnoconv("Account")), null, 'errors');
+if (empty($reshook)) {
+	// Selection of new fields
+	include DOL_DOCUMENT_ROOT.'/core/actions_changeselectedfields.inc.php';
+
+	// Purge search criteria
+	if (GETPOST('button_removefilter_x', 'alpha') || GETPOST('button_removefilter.x', 'alpha') || GETPOST('button_removefilter', 'alpha')) { // All tests are required to be compatible with all browsers
```

Comment:
from here to l.199, there is no change, it is only used in condition `if (empty($reshook))`